### DISCUSSION TITLE
Version 1.4.0-beta - Fields of Victory Support

### DISCUSSION THREAD

**Discussion by dashnine** [dashnine/PaperBlossoms#225] [General]
# Version 1.4.0-beta - Fields of Victory Support!

Thanks to a massive contribution from @MJKruszewski (thanks so much!), we are excited to announce support for Fields of Victory!  While we would normally leave a prerelease tag on this, I absolutely failed to remove the tag from prior builds (Celestial Realms!) for entirely too long. So: while we'll leave beta on the name, we're going to go ahead and designate this as a full release, as it contains a great deal to love. If you run into any bugs, please feel free to leave an issue!

## _Changelog (from 1.3.1):_
* **Now supports content from Deathly Turn**
* **Now supports content from Fields of Victory**

* _Corrected certain translation code corner cases_
* _Data bugfixes and typo corrections_
* _Corrected new character Ring calculation bugs_
* _Updated build files to support build on an M1_

As always, if you have custom user_* tables, be sure to use the 'Export' functionality to save your custom data.  When you have the opportunity to replace your database with a newer version, you'll need to do so (to handle the two new columns in the *curriculum* tables, in the event that you didn't grab 1.3.0--and to handle the new heritage data regardless).  You should be able to import prior data into the new table, however.

## TROUBLESHOOTING

Some users have reported that their database isn't getting updated when they first run the new application, resulting in empty curriculum tables, and a lack of choices on non-samurai new characters (or missing recent data such as the Centipede clan).  If this happens to you, you can resolve it using the following steps:

1. Tools->Open application Directory
2. You should see a file called paperblossoms -- that's the db.  Close the Paper Blossoms application.
3. Rename the paperblossoms.db file to something (paperblossoms.backup, for instance) and then start again.  It should regenerate the database and include all the new content.

This appears to most commonly be caused by something else having the DB open when you attempt an update, or by the prior DB being installed with different permissions (i.e. admin).  

<hr /><em>This discussion was created from the release <a href='https://github.com/dashnine/PaperBlossoms/releases/tag/v1.4.0-beta'>Version 1.4.0-beta - Fields of Victory Support</a>.</em>

**Comment by Jundiiii** [comment]
After creating and saving a new char, I tried to reload it.
The app returns an error message:

![image](https://github.com/user-attachments/assets/70872fba-2f93-4e77-8e75-a1066ad1c56b)

My settings.ini reports:

[General]
localeDB=it
savefilepath=F:/L5R/L5R 5ed/Personaggi
localeUI=it

How should I fix it?

Thank you in advance.

P.S. if needed, here the char file: 
[Kakita gooo.zip](https://github.com/user-attachments/files/17748288/Kakita.gooo.zip)

**Reply by dashnine** [reply]
I believe at this time, nobody has created a translation into IT, so a part of it is probably defaulting back to EN on launch.  Off the top of my head, I'm not sure why it's accepting the IT for one but not the other; you can probably file a bug for this one!  I have some changes that I've had in mind for a while; hopefully I can chase down a way to make that workable pretty fast if someone doesn't beat me to it :)

**Comment by BurnHeroSurge** [comment]
Hey, noticed a bug when using the special training function that when clicked causes the the program to crash out. Using a windows 11 pc with no direct modification but otherwise.
